packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol

```diff
@@ -263,27 +265,57 @@ abstract contract CrossDomainMessenger is
         // message is the amount of gas requested by the user PLUS the base gas value. We want to
         // guarantee the property that the call to the target contract will always have at least
         // the minimum gas limit specified by the user.
-        _sendMessage(
-            OTHER_MESSENGER,
-            baseGas(_message, _minGasLimit),
-            msg.value,
-            abi.encodeWithSelector(
-                this.relayMessage.selector,
-                messageNonce(),
-                msg.sender,
-                _target,
+
+
+        if (_type == BridgeConstants.BIT_TX || _type == BridgeConstants.ETH_WITHDRAWAL_TX) {
+            _sendMessage(
+                _type,
+                OTHER_MESSENGER,
+                baseGas(_message, _minGasLimit),
+                _amount,
+                abi.encodeWithSelector(
+                    this.relayMessage.selector,
+                    messageNonce(),
+                    msg.sender,
+                    _target,
+                    _amount,
+                    _minGasLimit,
+                    _message
+                )
+            );
+
+        } else if (_type == BridgeConstants.ETH_TX || _type == BridgeConstants.ERC20_TX) {
```

Comment:
How to handle `_sendMessage` if `_type` is `BIT_WITHDRAWAL_TX` or `ERC20_WITHDRAWAL_TX`